utilities/infra.py

```diff
@@ -583,10 +584,31 @@ def verify_no_failed_pods(
         isvc (InferenceService): InferenceService object
         runtime_name (str): ServingRuntime name
         timeout (int): Time to wait for the pod.
+        timeout_wait_for_pods (int): Time to wait for the pods.
+
     Raises:
-            FailedPodsError: If any pod is in failed state
+        FailedPodsError: If any pod is in failed state
+        TimeoutExpiredError: If no pods were created within `timeout_wait_for_pods` seconds.
 
     """
+    LOGGER.info("Wait for pods creation")
+
+    try:
+        for pods in TimeoutSampler(
+            wait_timeout=timeout_wait_for_pods,
+            sleep=10,
+            func=get_pods_by_isvc_label,
+            client=client,
+            isvc=isvc,
+            runtime_name=runtime_name,
+        ):
+            if pods:
```

Comment:
Should it matter how many pods you are expecting or as long as some pods exists, we can proceed with next step?

Reply:
no; at this stage have at least one is enough